packages/@expo/cli/src/export/exportAssets.ts

```diff
@@ -91,17 +139,41 @@ export async function exportAssetsAsync(
     (asset) => asset.hash
   );
 
+  let bundledAssetsSet: Set<string> | undefined = undefined;
+
   if (assets[0]?.fileHashes) {
+    debug(`Assets = ${JSON.stringify(assets, null, 2)}`);
+    // Updates the manifest to reflect additional asset bundling + configs
+    // Get only asset strings for assets we will save
+    bundledAssetsSet = resolveAssetPatternsToBeBundled(projectRoot, exp, assets);
+    let filteredAssets = assets;
+    if (bundledAssetsSet) {
+      debug(`Bundled assets = ${JSON.stringify([...bundledAssetsSet], null, 2)}`);
+      // Filter asset objects to only ones that include assetPatternsToBeBundled matches
+      filteredAssets = assets.filter((asset) =>
+        assetShouldBeIncludedInExport(asset, bundledAssetsSet)
+      );
+      debug(`Filtered assets count = ${filteredAssets.length}`);
+    }
     Log.log('Saving assets');
-    await saveAssetsAsync(projectRoot, { assets, outputDir });
+    await saveAssetsAsync(projectRoot, { assets: filteredAssets, outputDir });
   }
 
   // Add google services file if it exists
   await resolveGoogleServicesFile(projectRoot, exp);
 
-  // Updates the manifest to reflect additional asset bundling + configs
-  await resolveAssetBundlePatternsAsync(projectRoot, exp, assets);
-
+  bundles.ios?.assets.forEach((asset: Asset & { embedded?: true }) => {
```

Comment:
probably want to hoist this `embedded?: true` type into the `Asset` definition since this is mutating the parent object, so other methods that use this type need to know about the `embedded` field.

Reply:
I'll have a look -- since this only happens during the `export` command, most other methods would not be exposed to this.

Reply:
I see. I looked over the code and I think the cleaner way to do it is to keep the `embedded` logic out of the assets themselves.

So:
1. Make `exportAssetsAsync` return type also include `omittedAssets` or something like that. A record of which assets to exclude during `createMetadataJson` call.
2. Pass that into `createMetadataJson`. Then there's no need to modify the `Asset` typescript type (see failing `tsc`).

Reply:
Fixed